Automattic/wp-calypso
# Block Editor Tracking: Add editor_type and post_type props and rework tracking tests

#### Changes proposed in this Pull Request

* Add `editor_type` property to Gutenberg tracking events
  * `post` value is used when tracking the Post Editor
  * `site` value is used when tracking the Site Editor
* Add `post_type` property to Gutenberg tracking events
  * This property is only available in `edit-post` context (Post, Page, Widget, and Navigation Editor)
  * Value is the currently edited post's type
* Reworked Tracking E2E tests
  * Separated Post Editor, Page Editor, and Site Editor test files
    * `specs/specs-wpcom/wp-calypso-gutenberg-page-editor-tracking-spec.js`
    * `specs/specs-wpcom/wp-calypso-gutenberg-post-editor-tracking-spec.js`
    * `specs/specs-wpcom/wp-calypso-gutenberg-site-editor-tracking-spec.js`
  * Created a `general-tests` file that we run for each editor: `test/e2e/lib/gutenberg/tracking/general-tests.js`. We can specify editor-specific tests by adding them to the correct file.
* Fixed a bug where we were supposed to remove `undefined` values from the `eventProperties`. Lodash's `omit` function doesn't support `function` as an argument. We need to use `omitBy` in this case, as fixed in this PR:

#### Testing instructions

* Follow steps to enable tracking debugging at PCYsg-nrf-p2
* Make sure logged events contain the `editor_type` and `post_type` properties.
* Make sure properties contain the correct values.

**E2E tests should pass locally**
You can follow the steps here to configure your E2E tests env: PCYsg-o8X-p2

```bash
cd test/e2e

# Page editor
./node_modules/.bin/mocha specs/specs-wpcom/wp-calypso-gutenberg-page-editor-tracking-spec.js
# Post Editor
./node_modules/.bin/mocha specs/specs-wpcom/wp-calypso-gutenberg-post-editor-tracking-spec.js
# Site Editor
./node_modules/.bin/mocha specs/specs-wpcom/wp-calypso-gutenberg-site-editor-tracking-spec.js
```

#### TODO after merge

* [x] Update E2E instructions here: PCYsg-nrf-p2
* [ ] Create Site Editor Gutenberg Edge user



Related to #53411
Fixes https://github.com/Automattic/wp-calypso/issues/53411
